rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/expression/ExpressionGenerator.xtend

```diff
@@ -1176,4 +1178,63 @@ class ExpressionGenerator extends RosettaExpressionSwitch<JavaStatementBuilder,
 	override protected caseToZonedDateTimeOperation(ToZonedDateTimeOperation expr, Context context) {
 		conversionOperation(expr, context, '''«ZonedDateTime»::parse''', DateTimeParseException)
 	}
+
+	override protected caseSwitchOperation(SwitchOperation expr, Context context) {
+ 		val switchArgument = expr.argument.javaCode(MAPPER.wrap(typeProvider.getRType(expr.argument).toJavaReferenceType), context.scope)
+ 		val caseStatements = expr.values
+ 		val defaultExpression = expr.^default
+
+
+ 		val conditionType = switchArgument.expressionType.itemType
+
+ 		val returnType = MAPPER.wrap(join(
+ 				caseStatements.map[typeProvider.getRType(it.expression)] + (defaultExpression === null
+ 					? #[]
+ 					: #[
+ 					typeProvider.getRType(defaultExpression)
+ 				]
+ 			)
+ 		).toJavaReferenceType)
```

Comment:
Not sure if you actually need to wrap the return type in a mapper. In the common case I think this is not really necessary.

To have as less unnecessary wrapping/unwrapping as possible, perhaps you could try `val returnType = context.expectedType`?